pkg/controller/user-cluster-controller-manager/flatcar/flatcar_controller.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flatcar
+
+import (
+	"context"
+	"fmt"
+
+	"k8c.io/kubermatic/v2/pkg/controller/user-cluster-controller-manager/flatcar/resources"
+	nodelabelerapi "k8c.io/kubermatic/v2/pkg/controller/user-cluster-controller-manager/node-labeler/api"
+	predicateutil "k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "kubermatic_flatcar_controller"
+)
+
+type Reconciler struct {
+	ctrlruntimeclient.Client
+	overwriteRegistry string
+	updateWindow      kubermaticv1.UpdateWindow
+}
+
+func Add(mgr manager.Manager, overwriteRegistry string, updateWindow kubermaticv1.UpdateWindow) error {
+
+	reconciler := &Reconciler{
+		Client:            mgr.GetClient(),
+		overwriteRegistry: overwriteRegistry,
+		updateWindow:      updateWindow,
+	}
+
+	ctrlOptions := controller.Options{
+		Reconciler: reconciler,
+		// Only use 1 worker to prevent concurrent operator deployments
```

Comment:
Instead of doing it this way, you could use `pkg/controller/util.EnqueueConst("")` in the Watch().